### DISCUSSION TITLE
How to use the `History -> New` mode?

### DISCUSSION THREAD

**Discussion by hasezoey** [koreader/koreader#10701] [Q&A] [answered]
No matter what i do, i cannot get books to show up in `History -> New`, i have already done and checked:
- the calibre plugin and it says it has `146` books (via rescan library under `Search settings` (this weirdly contains 3 extra metadata about `Licenses`, `lastCrash` and `secondLastCrash`)
- (from the home / root book directory) the `+` in the file browser and `Extract and cache book information`, which says `143` books indexed (in koreader 2023.3 it also then said ~ `5` new indexed books)
- manually going to the folder where the new books are

history only says `Finished (129)`, `Reading (7)`, `New (0)` and `All (136)`
(the correct installed book count is `143`)

options used for `Extract and cache book information`

- Started from the book root directory (`/storage/sdcard1/Books/`)
- `Here and under`
- `Refresh` (also tried `Dont refresh` initially)
- `Prune` (also tried `Dont Prune` initially)

Device is `Tolino Vision 5` (android 4.4.2)

**Comment by poire-z** [comment]
From what I understand in the code, books with status "new" are books that don't have a metadata file (book.sdr/metadata.epub.lua).
But for a book to end up in History, it should be opened - and if opened, it will get a metadata file.
So, technically, I guess for a book in your History to be considered new again, you have to "Reset settings" it.
That is, "new" has no real sense in History :)
(Pinging @hius07 if I got that wrong.)

**Reply by hasezoey** [reply]
from my experience the books dont show up in anything (including `New`) with the following conditions:
- they are not indexed
- they are indexed
- they have a "new" metadata file

i just got one to show up, but only under the following condition:

- the book had to have metadata, had to be opened once and then has to be manually `Reset` (long press and click `Reset`)

---

maybe rename `History` to something along the lines of `Book Status` and actually add all indexed books?

**Reply by poire-z** [reply, marked as answer]
Our History is really the **history of user opened/read books**.
Files you put on your device won't go into History until you start reading them.
It is **not** a history of all the files one has put on the device: you will find them only in File browser.
(May be someday, if something like https://github.com/koreader/koreader/issues/8472#issuecomment-1552934881 happens, and is able to flatten your whole filesystem tree, and able to sort by book status, you'll get what I think you're wishing for.)

**Reply by hasezoey** [reply]
thanks, i was just wondering what the `New` tab was and if i recall correctly it was new in some 2023 version, but could never figure out how to get things in there (it made it seem like indexed books should go there and i was kinda hoping it would fulfill that purpose, because i dont remember which files i may have added but not read yet (and also not opened at least once) and also files that may have been updated (like changed paths) and so has different metadata)
